doltpy/etl/sql_sync/dolt.py

```diff
@@ -0,0 +1,119 @@
+from doltpy.core import Dolt
+from doltpy.etl.sql_sync.tools import TargetWriter, SourceReader
+from doltpy.etl.sql_sync.mysql import write_to_table as write_to_mysql_table, get_table_metadata
+import logging
+from typing import Iterable, List, Mapping
+
+logger = logging.getLogger(__name__)
+
+
+def get_source_reader(repo: Dolt, latest: bool = True, commit_ref: str = None, branch: str = None) -> SourceReader:
+    """
+    Returns a function that takes a list of tables
+    :param repo:
+    :param latest:
+    :param commit_ref:
+    :param branch:
+    :return:
+    """
+    def inner(tables: List[str]) -> Mapping[str, Iterable[tuple]]:
+        if branch is not None and branch != repo.get_current_branch():
+            repo.checkout(branch)
+        result = {}
+        repo_tables = repo.get_existing_tables()
+        missing_tables = [table for table in tables if table not in repo_tables]
+        if missing_tables:
+            logger.error('The following tables are missign, exiting:\n{}'.format(missing_tables))
+            raise ValueError('Missing tables {}'.format(missing_tables))
+
+        for table in tables:
+            logger.info('Syncing tables: {}'.format(tables))
+            result[table] = read_from_table(table, repo, latest, commit_ref)
+
+        return result
+
+    return inner
+
+
+def get_target_writer() -> TargetWriter:
+    raise NotImplemented()
+
+
+def read_from_table(table_name: str, repo: Dolt, latest: bool = True, commit_ref: str = None) -> List[tuple]:
+    """
+    This function pulls data from Dolt in three different modes:
```

Comment:
This smells funny. You have 2 bits of state and only 3 possible modes of operation. Yes, one mode throws an error, but still.

I would make this two functions: one for reading the whole table, and another for reading a diff, where ref = None means "give me the most recent diff"

Reply:
@zachmu good feedback, thanks